src/openrct2-ui/windows/ClearScenery.cpp

```diff
@@ -33,208 +33,169 @@ static constexpr const rct_string_id WINDOW_TITLE = STR_CLEAR_SCENERY;
+        {
+            widgets = window_clear_scenery_widgets;
+            enabled_widgets = (1ULL << WIDX_CLOSE) | (1ULL << WIDX_INCREMENT) | (1ULL << WIDX_DECREMENT)
+                | (1ULL << WIDX_PREVIEW) | (1ULL << WIDX_SMALL_SCENERY) | (1ULL << WIDX_LARGE_SCENERY) | (1ULL << WIDX_FOOTPATH);
+            hold_down_widgets = (1ULL << WIDX_INCREMENT) | (1ULL << WIDX_DECREMENT);
+            WindowInitScrollWidgets(this);
+            window_push_others_below(this);
+
+            gLandToolSize = 2;
+            gClearSceneryCost = MONEY64_UNDEFINED;
+
+            gClearSmallScenery = true;
+            gClearLargeScenery = false;
+            gClearFootpath = false;
+        }
+
+        void OnClose() override
+        {
+            if (clear_scenery_tool_is_active())
+                tool_cancel();
+        }
+
+        void OnMouseUp(rct_widgetindex widgetIndex) override
+        {
+            Formatter ft;
+            
+            switch (widgetIndex)
+                {
+                case WIDX_CLOSE:
+                    Close();
+                    break;
+                case WIDX_PREVIEW:
+                    ft.Add<int16_t>(MINIMUM_TOOL_SIZE);
+                    ft.Add<int16_t>(MAXIMUM_TOOL_SIZE);
+                    WindowTextInputOpen(WIDX_PREVIEW, STR_SELECTION_SIZE, STR_ENTER_SELECTION_SIZE, ft, STR_NONE, STR_NONE, 3);
+                    break;
+                case WIDX_SMALL_SCENERY:
+                    gClearSmallScenery ^= 1;
+                    Invalidate();
+                    break;
+                case WIDX_LARGE_SCENERY:
+                    gClearLargeScenery ^= 1;
+                    Invalidate();
+                    break;
+                case WIDX_FOOTPATH:
+                    gClearFootpath ^= 1;
+                    Invalidate();
+                    break;
+            }
+        }
+
+        void OnMouseDown(rct_widgetindex widgetIndex) override
+        {
+            switch(widgetIndex)
+                {
+                    case WIDX_DECREMENT:
+                        // Decrement land tool size, if it stays within the limit
+                        gLandToolSize = std::max(MINIMUM_TOOL_SIZE, gLandToolSize - 1);
+
+                        // Invalidate the window
+                        Invalidate();
+                        break;
+                    case WIDX_INCREMENT:
+                        // Increment land tool size, if it stays within the limit
+                        gLandToolSize = std::min(MAXIMUM_TOOL_SIZE, gLandToolSize + 1);
+
+                        // Invalidate the window
+                        Invalidate();
+                        break;
+                }
+        }
+
+        void OnTextInput(rct_widgetindex widgetIndex, std::string_view text) override
+        {
+            int32_t size;
+            char* end;
+
+            if (widgetIndex != WIDX_PREVIEW || text == nullptr)
+                return;
+
+            size = strtol(text.data(), &end, 10);
```

Comment:
This is not safe to perform you will need to convert the string view into a string. By doing this you are ignoring the length of the view.